pkg/cmd/render/render.go

```diff
@@ -0,0 +1,196 @@
+/*
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package render
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+
+	"github.com/ghodss/yaml"
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/kubeletconfig"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/machineconfig"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/runtimeclass"
+	"github.com/openshift-kni/performance-addon-operators/pkg/controller/performanceprofile/components/tuned"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+)
+
+type renderOpts struct {
+	ppInputFileDir string
+	assetsInDir    string
+	assetsOutDir   string
+}
+
+type manifestable map[string]v1.Object
+
+//NewRenderCommand creates a render command.
+func NewRenderCommand() *cobra.Command {
+	renderOpts := renderOpts{}
+
+	cmd := &cobra.Command{
+		Use:   "render",
+		Short: "Render performance-addon-operator manifests",
+		Long: `Render performance-addon-operator manifests creates the following manifests:
+		       - KubeletConfig.yaml
+			   - MachineConfig.yaml
+			   - RuntimeClass.yaml
+			   - Tuned.yaml`,
+		Run: func(cmd *cobra.Command, args []string) {
+
+			if err := renderOpts.Validate(); err != nil {
+				klog.Fatal(err)
+			}
+
+			if err := renderOpts.Run(); err != nil {
+				klog.Fatal(err)
+			}
+		},
+	}
+
+	renderOpts.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (r *renderOpts) AddFlags(fs *pflag.FlagSet) {
+	fs.StringVar(&r.ppInputFileDir, "performance-profile-dir-path", r.ppInputFileDir, "Input path for the Performance Profile manifest's directory.")
```

Comment:
do we need a separate directory? How other operators do?

Reply:
they provided separate directories if needed. see [here](https://github.com/openshift/installer/blob/master/data/data/bootstrap/files/usr/local/bin/bootkube.sh.template#L151)
Besides that, I don't think we need to force the user to have the performance-profile manifests together with other input-assets. This is more flexible don't you think?

Reply:
Looking at the example you provided, I see
```
		/usr/bin/cluster-config-operator render \
		--cluster-infrastructure-input-file=/assets/manifests/cluster-infrastructure-02-config.yml \
                ....
```
It seems to me a bit cleaner to just add something like
```
fs.StringVar(&r.performanceProfileInputFile, "performance-profile-input-file", r.performanceProfileInputFile, "Input path for the Performance Profile manifest.")
```
This way we get immediatley what we want, and we don't need anymore extra detection steps, e.g. with regexes.

Reply:
This was the original implementation, but we agreed to have the ability to render multiple pp files. 
Maybe I can provide a list of files` names,  instead of single file name or directory name

Reply:
Right. Having the option accept more than a single file (or having the operator accept the same option multiple times) seems the best option.

Reply:
How we can implement the same for env variable?
Is something like that:
```PERFORMANCE_PROFILE_INPUT_FILES=<file1's path>,<file2's path>,<file3's path>```
or ```PERFORMANCE_PROFILE_INPUT_FILES=<file1's path>:<file2's path>:<file3's path>```
is okay?

Reply:
yep, fine. Comma separated is a bit more common (while colon ":" is more uncommon) so I'd go for that.